### PR TITLE
Add support for YAML configuration file

### PR DESCRIPTION
**Pull request by williamfgc**

### CODE REVIEW

**Comment by williamfgc** [comment]
Feedback is welcome...

**Comment by suchyta1** [comment]
This looks good to me.

File: docs/user_guide/source/components/runtime.rst [hunk 1]
```diff
@@ -40,6 +47,57 @@ Each Engine should provide a set of available parameters as described in the :re
...
+     ...
+
+
+.. caution::
+   
+   YAML is case sensitive, make sure the node identifiers follow strictly the keywords: IO, Engine, Variables, Variable, Operations, Transports, Type.
+   
+.. tip::
+   
+   Run a YAML validator or use a YAML editor to make sure the provided file is YAML compatible.
```
[the remark this answers is not in the document]
Comment (williamfgc) [reply docs/user_guide/source/components/runtime.rst]: Any of the online validators would do the job. It happened to me in the past that I linked to tinypic for image hosting just to find out that it went away, better let the user pick their favorite.

File: source/adios2/core/ADIOS.cpp [hunk 2]
```diff
@@ -64,11 +64,15 @@ ADIOS::ADIOS(const std::string configFile, helper::Comm comm,
 {
     if (!configFile.empty())
     {
-        if (configFile.substr(configFile.size() - 3) == "xml")
+        if (helper::EndsWith(configFile, ".xml"))
```
[the remark this answers is not in the document]
Comment (williamfgc) [reply source/adios2/core/ADIOS.cpp]: C++17 would simplify a lot of the coding in adios2

File: source/adios2/helper/adiosYAML.cpp [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+        if (node && node.Type() != nodeType)
+        {
+            throw std::invalid_argument("ERROR: YAML: node " + nodeName +
+                                        " is the wrong type, review adios2 "
+                                        "config YAML specs for the node, " +
+                                        hint);
+        }
+    }
+    return node;
+}
```
[the remark this answers is not in the document]
Comment (williamfgc) [reply source/adios2/helper/adiosYAML.cpp]: No need for it in static workflows, where all user inputs, including the yaml structure, have been cleared.

File: thirdparty/yaml-cpp/Readme.txt [hunk 4]
```diff
@@ -0,0 +1,4 @@
+This is a copy of the upstream yaml-cpp library to parse and emit yaml files 
+located at https://github.com/jbeder/yaml-cpp.git. Do not make changes directly 
+to this repo but instead to the upstream repository.  Update this copy of
+yaml-cpp by running the update.sh script.
```
[the remark this answers is not in the document]
Comment (williamfgc) [reply thirdparty/yaml-cpp/Readme.txt]: Good question. It's not a strict copy, but a subset to reduce git and thirdparty footprint in libadios2, as long as the license allows us (e.g. MIT).

[the remark this answers is not in the document]
Comment (NAThompson) [reply thirdparty/yaml-cpp/Readme.txt]: Actually now that I think about it you're right: These git submodules often get really heavy.

**Pull request merged**